src/Arrays/MPIStateArrays.jl

```diff
@@ -352,7 +353,20 @@ function Base.copyto!(dest::MPIDestArray, src::MPIDestArray)
 end
 
 @inline function Base.copyto!(dest::MPIDestArray, bc::Broadcasted{Nothing})
-    copyto!(realview(dest), transform_broadcasted(bc, dest))
+    # check for the case a .= b, where b is an array
+    if bc.f === identity && bc.args isa Tuple{AbstractArray}
```

Comment:
I still think that this is fragile and encouraged the wrong code.

Reply:
Yeah, I tried to get everything to work out without this functionality over the weekend, but failed. Some initializations must be done on the CPU first (like when random numbers are involved).

I don't really like the CPU->GPU init method, so maybe we look into all reasons why such operations must be initialized on the CPU and fix them so they can be initialized on the GPU instead?

After that, it should be easy to remove this code.

Reply:
Right, or as elsewhere in the PR `dest .= src` needs to be rewriten as `copyto!(dest, src)`.

Reply:
This functionality has been adapted into a [commented function in the current PR](https://github.com/CliMA/ClimateMachine.jl/pull/1071/files#diff-0c102a9f3b330c12cac54a2a4821a48dR350). This function calls a [copyto method in GPUArrays](https://github.com/CliMA/ClimateMachine.jl/pull/1071/files#diff-0c102a9f3b330c12cac54a2a4821a48dR350) that copies twice, so it is probably a bit slow to use.

@mwarusz Could we chat to make sure all the copyto functionality is doing what it is intended to?

Reply:
Sure, thanks for cleaning up my broadcasting code. I believe that the only special functionality that I wanted to support
was to make MPIStateArrays broadcast only over the real elements (although it have never checked if this gives us any measurable performance benefit so maybe we should rethink this ?). I agree with Valentin that `copyto!` is cleaner for just transferring data.